User pipelines run endpoint for api trigger (#9133)
circleci/circleci-docs: jekyll/_cci2/triggers-overview.adoc

NOTE: Triggering a pipeline via the CircleCI v2 API is not currently supported for GitLab projects.

[.tab.trigger-pipeline-api.GitHub_and_Bitbucket]
--
This method is supported for GitHub (OAuth and App) and Bitbucket (Cloud and Data Center) orgs.

> TIP: For **projects integrated with GitHub OAuth app or Bitbucket Cloud**, you can also trigger a pipeline using the deprecated link:https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline[Trigger a New Pipeline] endpoint, which is described in the **CircleCI server** tab.

Reviewer: This is one those functionalities where support is determined by the type of **pipeline** rather than the type of org (org project). 
Can we say 
```suggestion
TIP: **GitHub OAuth and Bitbucket Cloud pipelines** can also be triggered using the deprecated link:https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline[Trigger a New Pipeline] endpoint, which is described in the **CircleCI server** tab.
```

Reviewer: Actually, Moses recommended not using the word "deprecated" until we have a clear deprecation plan, in order to avoid lots of support tickets being opened.

```suggestion
TIP: **GitHub OAuth and Bitbucket Cloud pipelines** can also be triggered using the link:https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline[Trigger a New Pipeline] endpoint, which is described in the **CircleCI server** tab.
```

And maybe even adding something like "but we recommend using the _recommended_ one"?
